src/components/Text/Annotated/AnnotatedText.tsx

```diff
@@ -2,50 +2,105 @@ import { BroccoliTextGeneric } from "../../../model/Broccoli.ts";
- * - Character index: start index marks first character to include, stop index marks first character to exclude
- * - Line segment: piece of line uninterrupted by annotation offsets
- * - Annotation segment: piece of an annotation uninterrupted by the offsets of other overlapping/nested annotations
- * - Annotation group: all annotations that are connected to each other by other overlapping/nested annotations
- * - Annotation depth: the number of levels that an annotation is nested in parent annotations or with overlapping annotations
+ * Annotation definitions
+ * - Depth: the number of levels that an annotation is nested in parent annotations or overlapping annotations
  *   (when two annotations overlap, the second annotation has a depth of 2)
+ * - Group: all annotations or segments that are connected to each other by other overlapping/nested annotations
+ * - Line: piece of annotated text as received from broccoli, a broccoli 'line' can also contain a logical text
+ * - Marker: annotation of zero length, useful for marking locations of footnotes, page endings, etc.
+ * - Offset: start or end character index of an annotation in a line
+ *   - start index marks first character to include
+ *   - stop index marks first character to exclude
+ * - Segment: piece of line or annotation uninterrupted by annotation offsets
  */
 export const AnnotatedText = (props: TextHighlightingProps) => {
+  const { footnoteMarkerAnnotationTypes, pageMarkerAnnotationTypes } =
+    useProjectStore(projectConfigSelector);
   const annotations = useAnnotationStore().annotations;
 
   const { tier2, highlight } = useDetailUrlParams();
   const searchTerms = highlight;
+  // const isDummy = DUMMY_ID === tier2;
+  const isDummy = false;
+  const annotationTypesToHighlight =
+    useAnnotationStore().annotationTypesToHighlight;
+  const lines = props.text.lines;
 
-  const typesToHighlight = useAnnotationStore().annotationTypesToHighlight;
-  const positions = props.text.locations.annotations;
+  // TODO: clean up dummy data
+  const relativeAnnotations = isDummy
+    ? [
+        {
+          bodyId: annotations.find((a) => a.body.type === "tei:Ptr")!.body.id,
+          start: { line: 0, offset: 10 },
+          end: { line: 0, offset: -1 },
+        },
+      ]
+    : props.text.locations.annotations;
+  // No offsets when no relative annotations
+  const offsets: LineOffsets[] = [];
 
-  const annotationsToHighlight = getAnnotationsByTypes(
-    annotations,
-    typesToHighlight,
-  );
-  const lines = props.text.lines;
+  const singleLineAnnotations = annotations.filter(withTargetInSingleLine);
+
+  if (relativeAnnotations.length) {
+    const nestedAnnotations = singleLineAnnotations
+      .filter((a) => annotationTypesToHighlight.includes(a.body.type))
+      .map((annotation) =>
+        createNestedLineOffsets(annotation, relativeAnnotations, lines),
+      );
+    console.log("nestedAnnotations", nestedAnnotations);
```

Comment:
Console log

Reply:
Fixed in [8988b27](https://github.com/knaw-huc/textannoviz/pull/154/commits/8988b2734d0fa14a4d9f8206dc51dc3b236b484f)